scanpy/_utils.py

```diff
@@ -546,6 +546,14 @@ def check_presence_download(filename: Path, backup_url):
         from .readwrite import _download
         _download(backup_url, filename)
 
+def check_presence_download_untar(filename: Path, tarfilename: Path, backup_url):
+    """Check if file is present otherwise download and untar."""
+    if not filename.is_file():
+        if not tarfilename.is_file():
+            from .readwrite import _download
+            _download(backup_url, tarfilename)
```

Comment:
Please avoid duplicating code. You can replace those three lines with `check_presence_download(tarfilename, backup_url)`

Reply:
Done.